src/python/gem5/components/processors/base_cpu_core.py

```diff
@@ -135,6 +135,11 @@ def connect_walker_ports(self, port1: Port, port2: Port) -> None:
     def set_workload(self, process: Process) -> None:
         self.core.workload = process
 
+    def is_workload_set(self) -> bool:
```

Comment:
It might be useful to move this function to `AbstractCore` so that anyone who makes their own cores would also have access to this function.